apps/chat-e2e/src/tests/sharedPromptFolderIcons.test.ts

```diff
@@ -0,0 +1,453 @@
+import { FolderInterface } from '@/chat/types/folder';
+import { Prompt } from '@/chat/types/prompt';
+import { ShareByLinkResponseModel } from '@/chat/types/share';
+import dialTest from '@/src/core/dialFixtures';
+import { ExpectedConstants, FolderPrompt, MenuOptions } from '@/src/testData';
+import { Colors } from '@/src/ui/domData';
+import { GeneratorUtil } from '@/src/utils';
+
+const nestedLevels = 3;
+
+dialTest(
+  'Shared icon appears in prompt folder if user open shared link.\n' +
+    'Share option appears in context menu for prompt folder if there is any prompt inside.\n' +
+    'Share form text differs for prompt and folder.\n' +
+    'Shared icon removes in prompt folder if rename prompt folder',
+  async ({
+    dialHomePage,
+    folderDropdownMenu,
+    promptBar,
+    shareModalAssertion,
+    folderPrompts,
+    folderDropdownMenuAssertion,
+    promptBarFolderAssertion,
+    promptData,
+    dataInjector,
+    additionalUserShareApiHelper,
+    confirmationDialog,
+    confirmationDialogAssertion,
+    setTestIds,
+  }) => {
+    setTestIds('EPMRTC-1816', 'EPMRTC-2731', 'EPMRTC-1817', 'EPMRTC-2817');
+    let nestedFolders: FolderInterface[];
+    const sharedFolderIndex = 2;
+    let shareResponse: ShareByLinkResponseModel;
+    const newFolderName = GeneratorUtil.randomString(5);
+
+    await dialTest.step(
+      'Prepare nested folders hierarchy and move prompt lowest level folder',
+      async () => {
+        nestedFolders = promptData.prepareNestedFolder(nestedLevels);
+        const lowLevelFolderPrompt = promptData.preparePromptInFolder(
+          GeneratorUtil.randomString(5),
+          GeneratorUtil.randomString(5),
```

Comment:
Why use random names if we could use something that makes sence and much easier to track and debug, such are names from ExpectedConstants + counter?

Reply:
utilizing random data in software testing is considered a good practice. This approach reflects the variety of input that can occur in a real-world scenario. Given that the entity name is unique, using ExpectedConstants+counter could cause confusion if an entity with the same name already exists in the app

Reply:
replaced with meaningful names